src/macros.js

```diff
@@ -233,6 +229,31 @@ defineMacro("\\thickspace", "\\;");   //   \let\thickspace\;
 //////////////////////////////////////////////////////////////////////
 // LaTeX source2e
 
+// \def\TeX{T\kern-.1667em\lower.5ex\hbox{E}\kern-.125emX\@}
+// TODO: Doesn't normally work in math mode because \@ fails.  KaTeX doesn't
+// support \@ yet, so that's omitted, and we add \text so that the result
+// doesn't look funny in math mode.
+defineMacro("\\TeX", "\\text{T\\kern-.1667em\\raisebox{-.5ex}{E}\\kern-.125emX}");
+
+// \DeclareRobustCommand{\LaTeX}{L\kern-.36em%
+//         {\sbox\z@ T%
+//          \vbox to\ht\z@{\hbox{\check@mathfonts
+//                               \fontsize\sf@size\z@
+//                               \math@fontsfalse\selectfont
+//                               A}%
+//                         \vss}%
+//         }%
+//         \kern-.15em%
+//         \TeX}
+// This code roughly aligns the top of the A with the T.  Here we approximate
+// this vertical alignment with a \raisebox{.45ex}, which seems close.
```

Comment:
What font size is being selected for the `A`?  Perhaps we could compute the `raisebox` amount based on the difference in size between the `A` and the `T`.

Reply:
The `A` should be in the equivalent of `\scriptsize`, which has a scale of 70%.  The height of a T is 0.68333em (according to `fontMetricsData.js`), so the raisebox should be `0.3 * 0.68333 = 0.204999`em.  I tried this, and it puts the A slightly higher than what I did, but it matches the LaTeX rendering better!  Good thinking.  I've pushed code that computes this automatically from the font data instead of hard-coding constants (other than 0.3 which seems safe enough).

Compare this screenshot with others on this page:
![katex](https://user-images.githubusercontent.com/2218736/33233323-94e5f952-d1e2-11e7-850a-e9717a68c514.png)